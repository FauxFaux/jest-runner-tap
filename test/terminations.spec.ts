import type { AssertionResult } from '@jest/test-result';
import { testTranslation } from './translate.spec';

describe('translation of terminations', () => {
  it('handles a test which timed out', async () => {
    const result = await testTranslation('./fixtures/timeout.tap');
    expect(result.numFailingTests).toBe(1);
    expect(result.numPendingTests).toBe(0);
    expect(result.numPassingTests).toBe(1);
    expect(result.skipped).toBe(false);
    expect(result.failureMessage).toMatchSnapshot();
    expect(result.testResults).toStrictEqual([
      expect.objectContaining({
        title: 'alpha',
        status: 'passed',
        numPassingAsserts: 1,
        failureMessages: [],
        ancestorTitles: [],
      } as Partial<AssertionResult>),
    ]);
    expect(result.testResults[0].duration).toBeGreaterThan(0);
  });
});
