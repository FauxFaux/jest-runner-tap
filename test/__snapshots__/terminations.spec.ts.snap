// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translation of terminations handles a test which timed out 1`] = `
"
  âœ•  bravo

    We failed to fully process the tap output:
    name: timeout!
    diag:
      signal: SIGTERM
      expired: TAP
      test: bravo
    fullname: timeout.js bravo
"
`;
